### PR TITLE
Readme changes

### PR DESCRIPTION
**Pull request by HelenaLykstoft**
- Changes to readme and adding images

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
 - Summarize your API’s performance based on the load testing results.
 - Identify areas where performance optimization may be needed.
 
----
+This project was a struggle to us. We had a hard time setting it up, and every time we pushed something that worked, it wouldnt work on other computers.
+We tried doing it as good as we could with the time we got, and we are happy with the result we got.
+If we were to make this project again, we would make sure we had enough code coverage, since we only made it to 52% of the overall coverage.
+
 
 ## Deliverables
```
Comment (thejamiegc) [inline ReadMe.md]: Burde hele afsnittet burde nok fjernes

File: ReadMe.md [hunk 2]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
----
 
 ## Reflection
 - Reflect on how you ensured code coverage.
+
+We ensured code coverage, by making sure not to write too many functions that we could not test.
+
 - Discuss the importance of balancing unit and integration testing.
 - Summarize your API’s performance based on the load testing results.
 - Identify areas where performance optimization may be needed.
```
Comment (thejamiegc) [inline ReadMe.md]: Ikke noget jeg mener burde være i opgaven. Det ligner punkter fra chat gpt

File: ReadMe.md [hunk 3]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
----
+- Program (81%): The code in the Program section has good test coverage (81%), meaning most of it has been exercised by integration tests.
+- Models and Controllers (100%): These sections are fully covered by tests, indicating that the behavior and structure of these components have been thoroughly tested.
+- Repositories and Data (0%): These sections have no test coverage. This is because the data layer only has our ToDoListContext, which is our data acccess layer for the database. We did not have time to implement tests for this layer.
 
-## API Testing
-- Use HTTP files (or Postman) to test all API endpoints.
-    - Ensure correct status codes (e.g., 200, 404, 500) are returned.
-    - Validate data returned in the response.
+Overall, we could have implemented more test coverage, but since we had a hard time with this project, it was not something we had time to do.
```
Comment (thejamiegc) [inline ReadMe.md]: Jeg kan ikke lige måden at formuleringen af den første sætning. 

Da jeg mener test coverage er ikke noget vi implementerer men man implementerer test til at have bedre code coverage.

File: ReadMe.md [hunk 4]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
 
-## REST API Development
-1. Build a simple REST API for a basic CRUD application (e.g., Task Manager).
-    - API should support:
-        - `POST /tasks` – Create a new task
-        - `GET /tasks` – Retrieve all tasks
-        - `GET /tasks/{id}` – Retrieve a specific task
-        - `PUT /tasks/{id}` – Update a specific task
-        - `DELETE /tasks/{id}` – Delete a task
+- **GetTask_ReturnsTaskById:**
```
Comment (thejamiegc) [inline ReadMe.md]: I alle metoderne skriver vi at det er et mock objekt. Uden er double checke er jeg i tvivl om det er et mock objekt.

File: ReadMe.md [hunk 5]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
-8. [Reflection](#reflection)
-9. [Deliverables](#deliverables)
+3. [JMeter Load Testing](#load-testing)
+4. [Reflection](#reflection)
 
 ---
 
 ## Overview
-This project involves building a simple CRUD (Create, Read, Update, Delete) REST API, writing unit and integration tests for the API, ensuring at least 80% code coverage, and performing basic performance tests with load testing tools.
+This project involves building a simple CRUD (Create, Read, Update, Delete) REST API, writing integration tests for the API and trying to ensuring at least 80% code coverage.
```
Comment (thejamiegc) [inline ReadMe.md]: Jeg synes "and try to ensure" lyder bedre.

File: ReadMe.md [hunk 6]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
-## Project Setup
-1. Clone this repository.
-2. Set up your development environment based on the language of choice.
-3. Install the necessary dependencies for unit testing, integration testing, and code coverage.
+## Testing and Coverage
+The things we managed to get done was:
+- Creating CRUD REST api, with possibility of testing via Swagger.
+- Creating integration tests for the API.
+- Ensuring (close to) 80% code coverage using Rider built-in test coverage.
+- Reflection
```
Comment (thejamiegc) [inline ReadMe.md]: Er det ikke det mere eller mindre det samme som der står oven over?

**Review by thejamiegc — CHANGES_REQUESTED** [review]

File: ReadMe.md (hunk 6 above)
Comment (solskinIsak) [reply ReadMe.md]: Jo, Tror bare vi enten skal beholde linjen der forklarer hvad vi nåede eller listen ikke begge to

File: ReadMe.md (hunk 4 above)
Comment (solskinIsak) [reply ReadMe.md]: uden at vide mig sikker så ligner det egentlig bare at` var mockTasks = new List<ToDoTask>`
et en variable der stadig henter en Liste af de reelle objekter så mener jeg ikke at objekterne bliver mocked

men servicen vi bruger sammen med de objekter er tilgengæld en mock 
`_mockTaskService = new Mock<ITaskService>();`


så tag mit bud med et gram salt men mener ikke objekterner er mocks men servicen der anvender dem i vores tests, er dermod mocks

File: ReadMe.md [hunk 7]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
-### Unit Tests
-- Write unit tests for the core logic of the API (e.g., task creation, validation, deletion).
-- Ensure the tests are comprehensive and check for edge cases (e.g., invalid input).
+- **GetTask_ReturnNotFound:**
+This test checks the behavior of the GetTask method when a non-existent task is requested. It mocks a task with ID 1, but queries the controller with ID 2, verifying that the result is a NotFound response.
 
-### Integration Tests
-- Write at least three integration tests that simulate real-world API interactions.
-    - Example: Test creating, updating, and retrieving a task through the API.
+Each test uses mock objects (Mock<ITaskService>) to simulate the service behavior without needing a live database, ensuring the controller logic is isolated and tested independently.
```
Comment (solskinIsak) [inline ReadMe.md]: Dette er rigtigt nok, så er dette `_mockTaskService = new Mock<ITaskService>();`  også hvad den referede til højere oppe ved jamies kommentar om det var mock objekter? 

hvis det er tilfældet så ignorer min kommentar til jamies

File: ReadMe.md (hunk 2 above)
Comment (HelenaLykstoft) [reply ReadMe.md]: Yes glemte at slette dem fra det vi havde af tekst i starten xD

File: ReadMe.md [hunk 8]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
 - Reflect on how you ensured code coverage.
+
+We ensured code coverage, by making sure not to write too many functions that we could not test.
+
 - Discuss the importance of balancing unit and integration testing.
 - Summarize your API’s performance based on the load testing results.
 - Identify areas where performance optimization may be needed.
 
----
+This project was a struggle to us. We had a hard time setting it up, and every time we pushed something that worked, it wouldnt work on other computers.
```
Comment (solskinIsak) [inline ReadMe.md]: Det er bedre at fokuserer på hvad vi nåede og hvad vi gjorde godt end at fokuserer på hvad vi ikke nåede og hvad der ikke virkede,

 intentionen her er rigtig god men tror det er bedre at pointerevi ikke nåede alle opgavens krav og så bare nævne i en reflektion hvad vi ville gøre anderledes en anden gang
 
"We didn't satisfy all the assignments criteria (or requirements) within the assigned timeslot and in order to still reach the deadline we focused on working with the tools and gathering an understanding for future projects" <-- agtig havde jeg nok skrevet 

og rundet af med det du skriver her:
`If we were to make this project again, we would make sure we had enough code coverage, since we only made it to 52% of the overall coverage.`
det her er helt fino som afrunding! ^

File: ReadMe.md [hunk 9]
```diff
@@ -8,101 +8,103 @@ The goal of this assignment is to build a simple REST API, implement tests for i
...
-    - Validate data returned in the response.
+Overall, we could have implemented more test coverage, but since we had a hard time with this project, it was not something we had time to do. 
 
-Include the HTTP test scripts (or Postman collection) in the project.
+### JMeter Load Testing
+![JMeterImg1.png](Images%2FJMeterImg1.png)
+![JMeterImg2.png](Images%2FJMeterImg2.png)
 
----
+This is what we got from the statistics:
```
Comment (solskinIsak) [inline ReadMe.md:67]: Godt arbejde at inkludere statistikkerne! det kan vi lide ! ^^

**Review by solskinIsak — APPROVED** [review]
Det ser helt fint ud for mig med rettelserne-  hvis jamie også er good med det så lad os vel bare fyre den af (～￣▽￣)～

**Review by thejamiegc — APPROVED** [review]
